tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -1535,7 +1532,47 @@
         float4 vColor2 = GetAttributeAtVertex(input.color, 2);
         return bary.x * vColor0 + bary.y * vColor1 + bary.z * vColor2;
       }
-      ]]>
+      ]]>          
+    </Shader>
+    
+    <!-- This part of the shader op is for the barycentrics ordering part of the barycentrics test -->
+    <Shader Name="PSordering" Target="ps_6_1" EntryPoint="PSMain" Text="@MAIN"/>
+    <Shader Name="VSordering" Target="vs_6_1" EntryPoint="VSMain" Text="@MAIN"/>
+
+    <Shader Name="MAIN" Target="vs_6_1" EntryPoint="VSMain">
+      <![CDATA[
+    struct PSInput {
+      float4 position : SV_POSITION;
+      nointerpolation float4 color : COLOR;
+      uint svid : SVertexID;
+    };    
+            
+    PSInput VSMain(float4 position : POSITION, float4 color : COLOR, uint svid : SV_VertexID) {
+      PSInput result;
+      result.position = position;
+      result.color = color;
+      result.svid = svid;
+         
+      return result;
+    }
+    
+    float4 PSMain(PSInput input) : SV_Target {
+      // check to make sure that the pixel shader will see the barycentric weight associated
+      // with the first vertex in the x component of the SV_Barycentric vector, and likewise for subsequent vertices
+      
+      if (0 == GetAttributeAtVertex(input.svid, 0) && 
+          1 ==  GetAttributeAtVertex(input.svid, 1) &&
+          2 ==  GetAttributeAtVertex(input.svid, 2))
+      {
+        // special value of all 1's in a float4 will be returned if the ordering is maintained for this pixel.
+        return float4(1.0, 1.0, 1.0, 1.0);
```

Comment:
An alternative way to do this (without the `if`) is:
````
return float4(float3(GetAttributeAtVertex(input.svid, 0),
                     GetAttributeAtVertex(input.svid, 1),
                     GetAttributeAtVertex(input.svid, 2)) * 0.5, 1.0)
```
then change the expected result to (0, 0.5, 1.0, 1.0) which makes out-of-order failure case reveal the ordering used in the result.